qa/tasks/cephfs/test_volumes.py

```diff
@@ -4212,6 +4204,84 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
 
+
+class TestSubVolumeRm(TestVolumesHelper):
+
+    def test_subvolume_rm_force(self):
+        # test removing non-existing subvolume with --force
+        subvolume = self._generate_random_subvolume_name()
+        try:
+            self._fs_cmd("subvolume", "rm", self.volname, subvolume, "--force")
+        except CommandFailedError:
+            self.fail("expected the 'fs subvolume rm --force' command to succeed")
+
+    def test_subvolume_rm_with_snapshots(self):
+        subvolume = self._generate_random_subvolume_name()
+        snapshot = self._generate_random_snapshot_name()
+
+        # create subvolume
+        self._fs_cmd("subvolume", "create", self.volname, subvolume)
+
+        # snapshot subvolume
+        self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
+
+        # remove subvolume -- should fail with ENOTEMPTY since it has snapshots
+        try:
+            self._fs_cmd("subvolume", "rm", self.volname, subvolume)
+        except CommandFailedError as ce:
+            if ce.exitstatus != errno.ENOTEMPTY:
+                raise RuntimeError("invalid error code returned when deleting subvolume with snapshots")
+        else:
+            raise RuntimeError("expected subvolume deletion to fail")
+
+        # remove snapshot
+        self._fs_cmd("subvolume", "snapshot", "rm", self.volname, subvolume, snapshot)
+
+        # remove subvolume
+        self._fs_cmd("subvolume", "rm", self.volname, subvolume)
+
+        # verify trash dir is clean
+        self._wait_for_trash_empty()
+
+    def _create_files_on_sv(self, sv_dir):
+        self.mount_a.remount(cephfs_name=self.volname)
+
+        self.mount_a.run_shell(f'sudo dd if=/dev/zero of={sv_dir}/file1 '
+                                'bs=1 count=0 seek=1G')
+        self.mount_a.run_shell(f'sudo cp {sv_dir}/file1 {sv_dir}/file2')
+        self.mount_a.run_shell(f'sudo cp {sv_dir}/file1 {sv_dir}/file3')
+
+    def _note_files_in_sv(self, sv_dir):
+        return self.mount_a.run_shell(f'sudo tree {sv_dir}').stdout.getvalue()
+
+    def assert_sv_is_present_in_trash(self, sv, sv_dir, sv_tree):
+        x = self.mount_a.run_shell('sudo ls volumes/_deleting').stdout.getvalue().strip()
+        trash_sv_dir = sv_dir.replace('_nogroup', f'_deleting/{x}')
+        trash_sv_dir = trash_sv_dir.replace(sv, '')
+        sv_tree_new = self.mount_a.run_shell(f'sudo tree {trash_sv_dir}').stdout.getvalue()
+        self.assertTrue(sv_tree_new, sv_tree)
+
+    def test_subvolume_rm_purge_trash(self):
+        '''
+        Test that when mgr config option mgr/volumes/purge_trash is set to
+        true, running "ceph fs subvolume rm" will move the subvolume to trash
+        but not delete the subvolume from trash.
+        '''
+        v, sv = self.volname, self._generate_random_subvolume_name()
+        self.run_ceph_cmd(f'fs subvolume create {v} {sv}')
+        self.run_ceph_cmd('config set mgr mgr/volumes/purge_trash false')
```

Comment:
nit: We should be adding an assert for `mgr/volumes/purge_trash` after using `config get` if we are changing the config option value

Reply:
Why is the check required after setting the config?

Reply:
If we are changing the config value in the tests, then we need to verify by getting it in my opinion.